mc/entity.go

```diff
@@ -120,10 +166,14 @@ func unmarshalEntityKey(key string, khash multihash.Multihash) (p2p_crypto.PubKe
 		return nil, err
 	}
 
-	hash := Hash(data)
+	return unmarshalEntityKeyBytes(data, khash)
+}
+
+func unmarshalEntityKeyBytes(key []byte, khash multihash.Multihash) (p2p_crypto.PubKey, error) {
+	hash := Hash(key)
 	if !bytes.Equal(hash, khash) {
```

Comment:
Minor thing: I would love to have a more futureproof interface around comparing multihashes, since the whole point is that they're self-describing. With `bytes.equal` we won't know whether the hash _value_ is different, or if the _hash algorithm_ is.